workflow/models.py

```diff
@@ -1,63 +1,198 @@
     description = models.TextField(null=False, blank=True)
     initialization_values = JSONField()
 
+    class JSONAPIMeta:
+        resource_name = 'workflows'
+
     def __str__(self):
         return self.title
 
+
 class Section(models.Model):
+    """Section Model"""
+
     id = models.AutoField(primary_key=True)
     label = models.CharField(max_length=128, blank=False, null=False)
     description = models.TextField(null=False, blank=True)
-    workflow = models.ForeignKey('Workflow', related_name='sections', null=False)
     index = models.IntegerField(null=True, blank=True)
+
+    workflow = models.ForeignKey(
+        'Workflow',
+        related_name='sections',
+        null=False
+    )
+
+    class JSONAPIMeta:
+        resource_name = 'sections'
+
     def __str__(self):
         return self.label
 
+
 class Widget(models.Model):
+    """Widget Model"""
+
     id = models.AutoField(primary_key=True)
     label = models.CharField(max_length=128, null=False, blank=False)
     description = models.TextField(null=False, blank=True)
     widget_type = models.CharField(max_length=24, blank=False, null=False)
-    parameter_mappings = models.ManyToManyField('WidgetParameterMapping', related_name='consumer_widgets')
-    section = models.ForeignKey('Section', related_name='widgets', null=False)
-    workflow = models.ForeignKey('Workflow', related_name='widgets', null=False)
     index = models.IntegerField(null=True, blank=True)
+
+    section = models.ForeignKey(
+        'Section',
+        related_name='widgets',
+        null=False
+    )
+
+    workflow = models.ForeignKey(
+        'Workflow',
+        related_name='widgets',
+        null=False
+    )
+
+    class JSONAPIMeta:
+        resource_name = 'widgets'
+
     def __str__(self):
         return self.label
 
-class WidgetParameterMapping(models.Model):
+
+class ParameterAlias(models.Model):
+    """ParameterAlias Model"""
+
     id = models.AutoField(primary_key=True)
-    name = models.CharField(max_length=64, blank=False, null=False)
-    parameter = models.ForeignKey('Parameter', related_name='widget_parameter_mappings', null=True)
-    case = models.ForeignKey('Case', null=True)
-    workflow = models.ForeignKey('Workflow', related_name='widget_parameter_mappings', null=False)
+    alias = models.CharField(max_length=64, blank=False, null=False)
+
+    widget = models.ForeignKey(
+        'Widget',
+        related_name='parameter_aliases',
+        null=True
+    )
+
+    parameter_stub = models.ForeignKey(
+        'ParameterStub',
+        related_name='aliases',
+        null=True
+    )
+
+    workflow = models.ForeignKey(
+        'Workflow',
+        related_name='parameter_aliases',
+        null=False
+    )
+
+    class JSONAPIMeta:
+        resource_name = 'parameter-aliases'
+
     def __str__(self):
-        return self.name
+        return self.alias
+
+
+class ParameterStub(models.Model):
+    """ParameterStub Model"""
```

Comment:
Widget stubs act as a placeholder for parameters. We can't clobber the parameters that are created for other cases. When a new case is started, we know what all the parameters should be and can create them.

Reply:
Stubs play no role in the context of a single case. We compute the parameters/aliases filtering for a particular case in the serializer so ember doesn't need to worry about it. This is why there's a parameters and a case/pk/parameters route.